### PR TITLE
fix batsat-ocaml build for macOS

### PR DESCRIPTION
**Pull request by bronsa**

### CODE REVIEW

File: src/batsat-ocaml/src/dune [hunk 1]
```diff
@@ -8,10 +8,11 @@
     (chdir ..
       (progn
        (run cp -r ../../../../src/batsat ..)
-       (chdir src (run cargo build --release))))
+       (setenv RUSTFLAGS "-L %{ocaml-config:standard_library} -lcamlrun"
+         (chdir src (run cargo build --release)))))
     ; TODO: strip the libs first
     (run cp ../target/release/libbatsat_stubs%{ext_lib} .)
-    (run cp ../target/release/libbatsat_stubs%{ext_dll} dllbatsat_stubs%{ext_dll})
+    (system "[ -f ../target/release/libbatsat_stubs.so ] && cp ../target/release/libbatsat_stubs.so . || cp ../target/release/libbatsat_stubs.dylib .")
```
Comment (c-cube) [inline src/batsat-ocaml/src/dune]: isn't `ext_dll` supposed to take care of the `.so` or `.dylib` distinction there?

Comment (bronsa) [reply src/batsat-ocaml/src/dune]:
```bash
[~]> ocamlc -config | grep ext_dll
ext_dll: .so
```
 ¯\\_(ツ)_/¯

Comment (c-cube) [reply src/batsat-ocaml/src/dune]: ôO ok…

**Comment by c-cube** [comment]
Now it fails on my side with

>   = note: /usr/bin/ld: /home/simon/.opam/4.06.0+flambda/lib/ocaml/libcamlrun.a(misc.o): relocation R_X86_64_PC32 against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
          /usr/bin/ld: final link failed: bad value
          collect2: error: ld returned 1 exit status


I notice that I used to rename `libbatsat_stubs.so` into `dllbatsat_stubs.so`, not sure if there's a difference. But I think in this case it's the explicit asking for `lcamlrun` that might fail…

**Comment by bronsa** [comment]
I  pushed a commit to reintroduce the renaming, not sure if it makes any difference but it seems like `dll` is the usual prefix for shared libraries so it doesn't hurt.. I'm trying to repro your issue on my machine

**Pull request merged**